prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -58,10 +78,181 @@ class Hl7Serializer(val metadata: Metadata) {
         outputStream.write(createFooters(report).toByteArray())
     }
 
+    /*
+     * Read in a file
+     */
+    fun convertBatchMessagesToMap(message: String, schema: Schema): Hl7Mapping {
+        val mappedRows: MutableMap<String, MutableList<String>> = mutableMapOf()
+        val errors = mutableListOf<String>()
+        val warnings = mutableListOf<String>()
+        val rowResults = mutableListOf<RowResult>()
+        val reg = "(\r|\n)".toRegex()
+        val cleanedMessage = reg.replace(message, "\r")
+        val messageLines = cleanedMessage.split("\r")
+        val nextMessage = StringBuilder()
+
+        fun deconstructStringMessage() {
+            println("Examining:\n$nextMessage")
+            val parsedMessage = convertMessageToMap(nextMessage.toString(), schema)
+            errors.addAll(parsedMessage.errors)
+            warnings.addAll(parsedMessage.warnings)
+            rowResults.add(parsedMessage)
+            nextMessage.clear()
```

Comment:
Because we are breaking down batch messages into single messages that start with the `MSH`, so I am putting the whole message into a `StringBuilder`